spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,124 @@
+require 'spec_helper'
+
+RSpec.describe ArticlesController do
+  describe 'GET index' do
+    describe '.html' do
+      it 'renders the index template and sets @articles' do
+        get :index
+        expect(assigns(:articles)).to eq([])
+        expect(response.status).to eq(200)
+        expect(response).to render_template("index")
+      end
+
+      describe 'with 20 articles' do
+        before(:each) { 20.times { create(:article, published: true) } }
+
+        it 'returns 10 articles in created at order' do
+          expected = Article.where(page: 1)
+                              .order(created_at: :desc)
+                              .limit(10)
+          get :index, page_id: 1
+          actual = assigns(:articles)
+          expect(actual.map(&:title)).to eq(expected.map(&:title))
+        end
+
+        it 'searches for articles' do
+          create(:article, published: true, title: 'hello')
+          get :index, search: 'hello'
+          expect(assigns(:articles).map(&:title)).to eq(['hello'])
+        end
+      end
+    end
+  end
+
+  describe 'GET show' do
+    before(:each) { create(:article, title: 'Mexican Revolution') }
+
+    it 'responds responds correctly' do
+      get :show, id: 1
+      expect(response.status).to eq(200)
+      expect(response).to render_template("show")
+    end
+
+    it 'shows an article' do
+      get :show, id: 1
+      expect(assigns(:article).title).to eq('Mexican Revolution')
+    end
+
+    it 'marks the article viewed' do
+      get :show, id: 1
+      expect(assigns(:article).impressionist_count).to eq(1)
+    end
+  end
+
+  describe 'GET new' do
+    before(:each) { sign_in :user, create(:user, role_ids: [1]) }
+    it 'returns an non-persisted article' do
+      get :new
+      expect(assigns(:article).persisted?).to be_falsey
+    end
+
+    it 'responds responds correctly' do
+      get :new
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe 'POST create' do
+    before(:each) { sign_in :user, create(:user, role_ids: [1]) }
+
+    it 'creates an article' do
+      post :create, article: attributes_for(:article)
+      expect(assigns(:article)).to eq(Article.first)
+    end
+
+    it 'assigns the expected params' do
+      params = attributes_for(:article)
+      post :create, article: params
+      attributes = assigns(:article).attributes
+      assigned_attributes = attributes.select { |attr|
+        params.keys.include?(attr.to_sym)
+      }
+      expect(assigned_attributes).to eq(params.stringify_keys)
+    end
+
+    it 'sets the expected defaults defaults' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.